build-dev-docker.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Prerequisites for releasing:
+
+# Logged in on Docker Hub (docker login)
+
+# fail this script immediately if any command fails with a non-zero exit code
+set -eu
+
+function push_failed {
+    echo "Error while pushing Docker image. Did you \`docker login\`?"
+}
+
+if [[ $# -ne 4 ]] ; then
+    echo "ERROR: $0 requires the Rally branch to build, the architecture, and whether to push the latest \
+        as command line arguments and they weren't supplied."
+    echo "For example: $0 master amd64 true true"
+    exit 1
+fi
+export RALLY_BRANCH=$1
+export ARCH=$2
+export PUSH_LATEST=$3
+export PUBLIC_DOCKER_REPO=$4
+
+if [[ $PUBLIC_DOCKER_REPO == "true" ]]; then
+    export DOCKER_IMAGE="elastic/rally"
+else
+    export DOCKER_IMAGE="docker.elastic.co/employees/es-perf/rally"
```

Comment:
@gbanasiak Is this fine? Or should we move it somewhere else?

Reply:
I think we should aim for `docker.elastic.co/rally` but can start from that.

Reply:
Alright, I'll leave it for now, until we get the TLD permissions to push there.